### PR TITLE
nvme/cfg: eliminate stale ctrls by using a finite ctrl_loss_tmo

### PR DESCRIPTION
**Pull request by anton-lb**
Until now we had a ctrl_loss_tmo set to -1 by default.

**This is likely a wrong lesson learnt from setups and tests with "manual" nvme connect's, where indeed ctrl_loss_tmo set lower than the maximal node downtime results in missing node connection later when the node comes back.**

This means a controller, once connected, is never removed, e.g. if a server is taken down/deleted,
the corresponding controllers will remain reconnecting forever on all client machines, until manually disconnected.

Solution:
- Add ctrlLossTMO to discovery-client.yaml (int, seconds, default is 600)
- Thus, any controller that cannot reconnect within this time will be automatically removed by nvme subsystem. NOTE: this won't take effect for existing controllers connected by an older discovery-client. The cluster discovery service notifies relevant discovery-clients when a node becomes active, so if a node is down and the corresponding controller is removed after 600s,
discovery-client will reconnect it once the node becomes active.

Consequences to expect:
1) If an server node is down and the corresponding controller is removed after 600s, discovery-client will reconnect it once the node becomes active: the cluster discovery service notifies relevant discovery-clients when a node becomes active.
2) If a client looses network connectivity to the servers, and the IO controllers are removed after 600s, But, discovery-client will also get disconnected and will keep trying to reconnect to the discovery endpoints. When network connectivity is restored, DC will succeed, fetch a log page,  and connect all the required IO controllers.
3) If a server is removed from the cluster, the discovery client configuration will still have this endpoint listed: discovery-client attempts to connect to it will fail to connect to a server that is gone or down, thus another endpoint will be picked.
4) If a client looses connection to all replicas of a volume (due to LB nodes being down or a network issue like (2), after 600s all IO controllers are removed, and the IO to the volume will fail ("no path available - failing IO). Before this change the IO would hang forever without failing (possible path present). The ctrlLossTMO in the configuration file can be used to mitigate this if needed.

** How this PR was tested **
- Manually verified the discovery connects a node after it becomes active again
- Tests: 529, 279

Issue: LBM1-37762

### CODE REVIEW

**Review by jeffry-lb — APPROVED** [review]
@anton-lb smart idea, the changes themselves look good, but I'm wondering about a few corner cases.

1. For whatever reason, a client is disconnected from a target for longer than 10 minutes; the controller stops reconnecting, but it has volumes present. Today, it would eventually reconnect without an AEN event firing. With this chang manual intervention required?
2. This does not really remove the target controller from the configuration, and during restarts, etc., connection attempts to a dead target might still show up since the referrals are stored in the cache and delay data availability (we first timeout to the removed target, before picking the next one). 

Thoughts?

File: pkg/nvmeclient/nvme_client.go [hunk 1]
```diff
@@ -620,7 +620,7 @@ func ConnectAllNVMEDevices(logPageEntries []*hostapi.NvmeDiscPageEntry,
 		if logPageEntry.SubType != nvme.NVME_NQN_NVME {
 			continue
 		}
-		ctrlLossTMOValue := -1
+		ctrlLossTMOValue := cfg.CtrlLossTMO
 		if ctrlLossTMO != nil {
 			ctrlLossTMOValue = *ctrlLossTMO
 		}
```
[the remark this answers is not in the document]
Comment (sanderdescamps) [reply pkg/nvmeclient/nvme_client.go:626]: I'm having this exact issue. @anton-lb please look again at the original comment. `cfg.CtrlLossTMO` can't be `nil` indeed but `cfg` can. Which causes a runtime error.
